text_formatter.go

```diff
@@ -154,11 +154,12 @@ func (f *TextFormatter) needsQuoting(text string) bool {
 }
 
 func (f *TextFormatter) appendKeyValue(b *bytes.Buffer, key string, value interface{}) {
-
+	if b.String() != "" {
```

Comment:
This is going to cost allocation of the whole buffer. If this is used in a loop, a string will be created when appending each value, generating a lot of garbage for every log message. It would be better to test the length of the buffer.